localstack/http/request.py

```diff
@@ -278,6 +278,9 @@ def restore_payload(request: Request) -> bytes:
       inevitably become the the basis of proxying werkzeug requests. The alternative is to build our own request object
       that memoizes the original payload before parsing.
     """
+    if request.shallow:
+        return b""
```

Comment:
perhaps it would be better to raise an exception here instead of returning empty bytes, but this "just works" (it's mostly to fix the request logger)